edge-util/src/Microsoft.Azure.Devices.Edge.Storage.RocksDb/ColumnFamilyDbStore.cs

```diff
@@ -75,7 +75,20 @@ public async Task Remove(byte[] key, CancellationToken cancellationToken)
 
             Action operation = () => this.db.Remove(key, this.Handle);
             await operation.ExecuteUntilCancelled(cancellationToken);
-            Interlocked.Decrement(ref this.count);
+            // We don't want to wrap around the counter.
+            var currentCount = Interlocked.Read(ref this.count);
```

Comment:
You will need to lock the entire block - if 3 threads read a value (say 2) and all 3 call decrement you will end up with -1.